static/app/views/automations/components/editConnectedMonitors.tsx

```diff
@@ -0,0 +1,80 @@
+import {Fragment, useState} from 'react';
+import styled from '@emotion/styled';
+
+import SearchBar from 'sentry/components/searchBar';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {Detector} from 'sentry/types/workflowEngine/detectors';
+import ConnectedMonitorsList from 'sentry/views/automations/components/connectedMonitorsList';
+
+export default function EditConnectedMonitors() {
+  const {monitors, connectedMonitorIds, toggleConnected} = useConnectedMonitors();
+
+  const connectedMonitors = monitors.filter(monitor =>
+    connectedMonitorIds.has(monitor.id)
+  );
+  const unconnectedMonitors = monitors.filter(
+    monitor => !connectedMonitorIds.has(monitor.id)
+  );
+
+  return (
+    <div>
+      {connectedMonitors.length > 0 && (
+        <Fragment>
+          <Heading>{t('Connected Monitors')}</Heading>
+          <ConnectedMonitorsList
+            monitors={connectedMonitors}
+            connectedMonitorIds={connectedMonitorIds}
+            toggleConnected={toggleConnected}
+          />
+        </Fragment>
+      )}
+      <Heading>
+        {connectedMonitors.length > 0 ? t('Other Monitors') : t('All Monitors')}
+      </Heading>
+      <div style={{flexGrow: 1}}>
+        <StyledSearchBar placeholder={t('Search for a monitor or project')} />
+      </div>
+      <ConnectedMonitorsList
+        monitors={unconnectedMonitors}
+        connectedMonitorIds={connectedMonitorIds}
+        toggleConnected={toggleConnected}
+      />
+    </div>
+  );
+}
+
+export function useConnectedMonitors() {
+  // TODO: Fetch monitors from API
+  const monitors: Detector[] = [];
```

Comment:
Two thoughts on this abstraction:
1. Keep in mind that this connections interaction is needed for both automations + monitors, so it'd be great to keep this more generic so it supports both.
2. When editing an existing workflow, we'll want to pass in the connected ids to set the initial state.

Reply:
both good points, i think i'll refactor this to support both types (automations + monitors) once we have the automation types merged and i've refactored the `connectedAutomationsList`

Reply:
Sounds great to me!